.github/workflows/main.yml

```diff
@@ -0,0 +1,177 @@
+        fetch-depth: 0
+        repository: HEXRD/hexrd
+        path: hexrd
+
+    - name: Install conda
+      uses: goanpeca/setup-miniconda@v1
+      with:
+        auto-update-conda: true
+        python-version: 3.8
+        auto-activate-base: false
+
+    - name: Get version using git describe
+      working-directory: hexrdgui
+      id: hexrdgui_describe
+      run: echo "::set-output name=version::$(git describe --tag)"
+
+    - name: Conda environment to build HEXRD master ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      #working-directory: hexrd
+      run: |
+          conda create --name hexrd python=3.8
+      # This is need to ensure ~/.profile or ~/.bashrc are used so the activate
+      # command works.
+      shell: bash -l {0}
+
+    - name: Install build requirements for HEXRD ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      #working-directory: hexrd
+      run: |
+          conda activate hexrd
+          conda install anaconda conda-build
+      # This is need to ensure ~/.profile or ~/.bashrc are used so the activate
+      # command works.
+      shell: bash -l {0}
+
+    - name: Build the HEXRD conda package ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      working-directory: hexrd
+      run: |
+          conda activate hexrd
+          mkdir output
+          conda build --output-folder output/ conda.recipe/
+          pwd
+      # This is need to ensure ~/.profile or ~/.bashrc are used so the activate
+      # command works.
+      shell: bash -l {0}
+
+    - name: Get HEXRD version using git describe ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      working-directory: hexrd
+      id: hexrd_describe
+      run: echo "::set-output name=version::$(git describe --tag)"
+
+    - name: Upload the HEXRD conda package ( PRs only )
+      if: github.ref != 'refs/heads/master'
+      #working-directory: hexrd
+      uses: actions/upload-artifact@v2
+      with:
+        name: HEXRD-${{ matrix.config.name }}-${{ steps.hexrd_describe.outputs.version }}.tar.bz2
+        path: ${{ github.workspace }}/hexrd/output/**/*.tar.bz2
+
+    - name: Conda environment to build HEXRDGUI
+      working-directory: hexrdgui
+      run: |
+          conda env create -f packaging/environment.yml
+      # This is need to ensure ~/.profile or ~/.bashrc are used so the activate
+      # command works.
+      shell: bash -l {0}
+
+    - name: Set version environment variables used by CPack
+      working-directory: hexrdgui
+      run: |
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} full`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} major`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} minor`
+          echo `python packaging/github_action_version.py ${{ steps.hexrdgui_describe.outputs.version }} patch`
```

Comment:
I think these don't need to be "echo'ed". I did something similar [here](https://github.com/hydroframe/SandTank/blob/b1a6d50ba33187ab1fcb54c760ba71fa1cfb3ab4/.github/workflows/docker.yml#L50), where the script itself sets a variable, and I don't think an echo was needed when calling the script.

Reply:
But it's a python script?

Reply:
I think github actions just checks all of the stdout that comes out of the `run` command, and if anything starts with `::set-env`, it considers it to be telling it to set a variable.

Reply:
Or `::set-output`

Reply:
Ok, thanks, will fix that up